assets/JS/index.js

```diff
@@ -0,0 +1,117 @@
+// Assignment Code
+const generateBtn = document.querySelector("#generate");
+//arrays used to generate PW
+const alphaLower = ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "n", "o", "p", "q", "r", "s", "t", "u", "v", "w", "x", "y", "z"];
+const alphaUpper = ["A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z"];
+const numbers = ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9"];
+const special = ["!", '"', "#", "%", "&", "'", "(", ")", "*", "+", "," ,"-",".","/",":",";","<","=",">","?","@","[","]","^","_","`","{","|","}","~"];
+
+// main function to generate the random password
+const generatePassword = () => {
+  
+
+// get the password length
+const passwordLength = getPasswordLength();
+
+if (passwordLength) {
+// get the password criteria
+  const passwordCriteria = getPasswordCriteria();
```

Comment:
Your `getPasswordCriteria()` function returns an empty array if no options are chosen. Therefore, we get a problem here when we pass an empty array to the `createRandomPassword` funciton on line 26. An error appears in teh console because you are still running `createRandomPassword` even if no choices have been selected. You need to control the flow of your code here and only proceed to run `createRandomPassword` IF `passwordCriteria.length` is truthy.